testkit/testmain/testmain.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !codes
+
+package testmain
+
+import "go.uber.org/goleak"
+
+type testingM struct {
+	goleak.TestingM
+	callback func(int) int
+}
+
+func (m *testingM) Run() int {
+	return m.callback(m.TestingM.Run())
+}
+
+// WrapTestingM returns a TestingM wrapped with callback on m.Run returning
+func WrapTestingM(m goleak.TestingM, callback func(int) int) *testingM {
```

Comment:
Is it not used?

Reply:
Used here https://github.com/pingcap/tidb/pull/27743/files#diff-48dd7515eec4786192bae5129ca431ce9ce97d09acf86b2ac2defd1616136022R39

Workaround `goleak.VerifyMain(m)` shortcoming on call back after `m.Run`.